core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/InvokerReactive.scala

```diff
@@ -244,7 +244,8 @@ class InvokerReactive(
                     ActivationResponse.applicationError(Messages.actionRemovedWhileInvoking)
                   case _: DocumentTypeMismatchException | _: DocumentUnreadable =>
                     ActivationResponse.whiskError(Messages.actionMismatchWhileInvoking)
-                  case _ =>
+                  case e =>
+                    logging.error(this, s"${Messages.actionFetchErrorWhileInvoking} Error: ${e.getMessage}")
```

Comment:
Thanks for this helpful addition. I suggest to cover also the other two `case`s because these are system errors as well.

Reply:
Isn't the error already printed by the database layer? If not: Should it be put there?

Reply:
@markusthoemmes are you talking about the `ArtifactStore` implementations - for example: https://github.com/apache/incubator-openwhisk/blob/master/common/scala/src/main/scala/org/apache/openwhisk/core/database/CouchDbRestStore.scala#L217?

If yes - I missed that we already log information there.

Honestly, I don't understand what's going on here: https://github.com/apache/incubator-openwhisk/blob/master/common/scala/src/main/scala/org/apache/openwhisk/core/database/CouchDbRestStore.scala#L242. First, we finished the transaction and later on via `reportFailure()`, we fail it because an exception is thrown?

Reply:
Every error in the artifact store is already logged as Markus noted. reportFailure is there to make sure the error isn’t lost. It doesn’t modify the future.

Reply:
https://github.com/apache/incubator-openwhisk/blob/be1e3a19c02956c9be85023a0bb0ff399c21444d/common/scala/src/main/scala/org/apache/openwhisk/core/database/StoreUtils.scala#L37

All failured future get logged there. So the earlier change above is redundant isn’t it?

Reply:
@rabbah You are right. I missed that... Can you please have a look here?

My understanding is that we have the following flow:
1. `Left(code)` matches.
2. `transid.finished()` is called.
3. A new exception is thrown, failing Future `f`.
4. `reportFailure()` is called with the failed `f`. Error handling calls `transid.failed()`.

Do your agree? What's the rationale behind finishing the transaction first and then failing it?

https://github.com/apache/incubator-openwhisk/blob/a08175f05cd9bf21e48eea52be62ca0030f9c2c3/common/scala/src/main/scala/org/apache/openwhisk/core/database/CouchDbRestStore.scala#L242-L258

Reply:
your 1-4 is correct... but I don't understand the question about `transid.failed` and specifically "failing it". The `reportFailure` method call to `transid.failed` only serves to log the failed future: https://github.com/apache/incubator-openwhisk/blob/a92d14c3cd685ec8c9caa530606a82bb9aed9405/common/scala/src/main/scala/org/apache/openwhisk/common/TransactionId.scala#L137

Reply:
Well, on the same `transid`, we first call `finished()` and then `failed()`. So for one single `transid`, we have two log entries - one indicating success and a second indicating failure. In addition, two metrics are sent. Correct?

Reply:
thanks for conferring on slack - we agree there is duplication: the `finished` message is logged at `InfoLevel` and `failed` at `ErrorLevel` and both of these generate metrics. The future is not modified in either case.

Given the duplication, and if all the paths in the future are already logging, there is less of a need for `reportFailure`. So I think that should be addressed. With the change @mhenke1 made, now `failed` will be called twice on failure. Further I think the change in the Invoker is not necessary and redundant.

Now back to the existential question: if we agree there should have been an Error level message reported anyone on the failed future, what were you missing from the logs?